src/typescript/typescript-visitor.ts

```diff
@@ -0,0 +1,747 @@
+        break;
+      case ts.SyntaxKind.PropertyAccessExpression:
+        this.visitPropertyAccessExpression(<ts.PropertyAccessExpression>node);
+        break;
+      case ts.SyntaxKind.PropertyAssignment:
+        this.visitPropertyAssignment(<ts.PropertyAssignment>node);
+        break;
+      case ts.SyntaxKind.PropertyDeclaration:
+        this.visitPropertyDeclaration(<ts.PropertyDeclaration>node);
+        break;
+      case ts.SyntaxKind.PropertySignature:
+        this.visitPropertySignature(node);
+        break;
+      case ts.SyntaxKind.RegularExpressionLiteral:
+        this.visitRegularExpressionLiteral(node);
+        break;
+      case ts.SyntaxKind.ReturnStatement:
+        this.visitReturnStatement(<ts.ReturnStatement>node);
+        break;
+      case ts.SyntaxKind.SetAccessor:
+        this.visitSetAccessor(<ts.AccessorDeclaration>node);
+        break;
+      case ts.SyntaxKind.SourceFile:
+        this.visitSourceFile(<ts.SourceFile>node);
+        break;
+      case ts.SyntaxKind.StringLiteral:
+        this.visitStringLiteral(<ts.StringLiteral>node);
+        break;
+      case ts.SyntaxKind.SwitchStatement:
+        this.visitSwitchStatement(<ts.SwitchStatement>node);
+        break;
+      case ts.SyntaxKind.TemplateExpression:
+        this.visitTemplateExpression(<ts.TemplateExpression>node);
+        break;
+      case ts.SyntaxKind.ThrowStatement:
+        this.visitThrowStatement(<ts.ThrowStatement>node);
+        break;
+      case ts.SyntaxKind.TryStatement:
+        this.visitTryStatement(<ts.TryStatement>node);
+        break;
+      case ts.SyntaxKind.TupleType:
+        this.visitTupleType(<ts.TupleTypeNode>node);
+        break;
+      case ts.SyntaxKind.TypeAliasDeclaration:
+        this.visitTypeAliasDeclaration(<ts.TypeAliasDeclaration>node);
+        break;
+      case ts.SyntaxKind.TypeAssertionExpression:
+        this.visitTypeAssertionExpression(<ts.TypeAssertion>node);
+        break;
+      case ts.SyntaxKind.TypeLiteral:
+        this.visitTypeLiteral(<ts.TypeLiteralNode>node);
+        break;
+      case ts.SyntaxKind.TypeReference:
+        this.visitTypeReference(<ts.TypeReferenceNode>node);
+        break;
+      case ts.SyntaxKind.VariableDeclaration:
+        this.visitVariableDeclaration(<ts.VariableDeclaration>node);
+        break;
+      case ts.SyntaxKind.VariableStatement:
+        this.visitVariableStatement(<ts.VariableStatement>node);
+        break;
+      case ts.SyntaxKind.WhileStatement:
+        this.visitWhileStatement(<ts.WhileStatement>node);
+        break;
+      case ts.SyntaxKind.WithStatement:
+        this.visitWithStatement(<ts.WithStatement>node);
+        break;
+      default:
+        console.warn(`unknown node type: ${node}`);
+        break;
+    }
+  }
+
+  visitChildren(node: ts.Node): void {
+    ts.forEachChild(node, (child) => this.visitNode(child));
+  }
+
+  visitAnyKeyword(node: ts.Node): void {
+    this.visitChildren(node);
```

Comment:
I'm not fully parsing what you're saying here. Are you suggesting to inline the visitChildren implementation into the visit* methods?